lib/vty.c

```diff
@@ -2207,6 +2235,34 @@ static int vtysh_do_pass_fd(struct vty *vty)
 	return BUFFER_EMPTY;
 }
 
+/* This function is similar to vtysh_flush exceot this is used
+ * to chunkify the vty_outs intermediately when the buffer size
+ * is 128KB or more (since only 131 KB is flushed at once).
+ *
+ * When a BUFFER_EMPTY is received, we clean up the buffer data
+ * When a BUFFER_PENDING is received, we schedule to call the same
+ * function again to process the data in the pending buffer
+ */
+static void vtysh_flush_intermediate(struct vty *vty)
```

Comment:
so ... this is a lot like "vtysh_flush()" - why not just ... flush periodically? why should the behavior of "periodically flush" be different or special - you want to call "buffer_flush_available()", and then move on?

Reply:
We want to handle `BUFFER_PENDING` and `BUFFER_EMPTY` differently compared to `vtysh_flush`. Specifically, when `BUFFER_EMPTY` occurs, we want to clear the allocated memory, which `vtysh_flush` does not currently do. For `BUFFER_PENDING`, we will take appropriate actions to either flush the buffer or continue processing as needed.

Reply:
uh  - but isn't `buffer_flush_available()` already doing that - it looks like it frees whatever it was able to send. that would be ... everything, if it returns "EMPTY" ?

Reply:
Yes, thanks for pointing that out. I have made changes to use existing `vtysh_flush()` function and no new function to intermediately call for flushing the data.